src/providers/scheduling-service.ts

```diff
@@ -199,8 +199,13 @@ export class SchedulingService {
       taskPromise = this.getClinicalTasks()
     }
     return taskPromise.then(tasks => {
-      const updatedTasks = tasks
-      updatedTasks[task.index] = task
+      const updatedTasks = tasks.map(d => {
+        if (d.index == task.index) {
```

Comment:
Maybe `===` instead of `==`. Although we know these will always be of same type but for the sake of consistency.

Reply:
Also can be converted to a single line `return d.index === task.index ? task : d`